### DISCUSSION TITLE
I (almost) destroyed a UGREEN RTL8156BG adapter so you don't have to :)  (DS418play)

### DISCUSSION THREAD

**Discussion by DrmnSamoLiu** [bb-qq/r8152#383] [Show and tell]
**Hardware:**
DS418play

**DSM:** 
DSM 7.2.1-69057 Update 1

**Driver:** 
r8152-apollolake-2.17.1-1_7.2.spk 

**Status:**

Work

**Notes:**

1. After driver package installation, I realized I have to manually enable it from package manager.
2. I can't "plug and play" the adapter, the most stable way to get the adapter recognized as network interface is to disable then enable again the driver package while the adapter is plugged in.

**Discussion:**

Although iperf3 test results are great, I can only get around 180 MB/s SMB download, and 110~130MB/s SMB upload speed on Win10, while HTTP download can easily hit 200MB/s.
I was downloading from a encrypted share folder to a bitlocker enabled HDD, so I guess 200MB/s is a reasonable max speed I could get.
Why SMB is so slow still puzzles me though, any ideas are well appreciated.

--------

Today I received my UGREEN adapter with RTL8156BG chipset in it. As I'm curious about its thermals, I decided to break it open and do some tests along with network performance tests.

Here are some photos of it:

--------
Outer case, with model name on it.
I made a stupid mistake trying to open it up from the RJ45 connector side (and destroyed the plastic part), while it can be easy popped out from the cable side.....<br>
![DSC05614](https://github.com/bb-qq/r8152/assets/36998819/d48197ed-ae46-4b9d-a6ef-807d825d7399)

--------
Overview of the circuit board, it is way simpler than I thought, and there are no heat sinks or anything that is used to dissipate heat.<br>
 ![DSC05599](https://github.com/bb-qq/r8152/assets/36998819/c109d99c-512f-497b-99ab-6bd3b06ba27c)

--------
The RTL8156BG chip itself, with the special Realtek DRAGON icon on it, instead of the standard "crab" icon.<br>
![DSC05611](https://github.com/bb-qq/r8152/assets/36998819/bfdd742b-4649-44ca-b916-fd13b355a477)

--------

20 mins iperf3 test have stable 2.5Gbps results:
![圖片](https://github.com/bb-qq/r8152/assets/36998819/957c54cc-f476-42eb-a325-d3ad554ff47f)

The chip thermal is basically capped at around 45c:
(Tested with both a temp sensor and IR camera, room temp 22c)
![PXL_20240305_182516295 NIGHT](https://github.com/bb-qq/r8152/assets/36998819/5813e009-1cc2-46cf-89a4-98c95c616ab7)
![img_thermal_1709663013649](https://github.com/bb-qq/r8152/assets/36998819/46bf0d43-8ec2-445f-864f-a24bb6fb56d2)

 The temperature might hit 50c if it's still in the case.
 It's not awfully hot but I personally would consider adding a tiny heat sink to it when transferring files that will take a long time, especially when the weather gets hotter.

**Comment by ydzhus** [comment]
![зображення_viber_2024-06-15_14-23-26-040](https://github.com/bb-qq/r8152/assets/141327954/d8f42cc6-449e-461e-8ea2-51a712831b64)
![зображення_viber_2024-06-15_14-23-26-279](https://github.com/bb-qq/r8152/assets/141327954/6ee61c7c-bd4e-4eda-94eb-2758d91c3c38)
![зображення_viber_2024-06-15_14-23-26-499](https://github.com/bb-qq/r8152/assets/141327954/31975000-e623-4cc0-ae50-554a3b39123c)
![зображення_viber_2024-06-15_14-23-26-714](https://github.com/bb-qq/r8152/assets/141327954/1ba5bae4-3ddb-4349-82b4-cbd7b1926eab)

**Reply by nxxxse** [reply]
@ydzhus How did you open it? Can it be opened without breaking it? I tried but the USB side plastic sits in firmly. I can bend it only so that a tiny gap between plastic an aluminum opens.

**Reply by ydzhus** [reply]
opens from the usb side with a flat blade or knife, but I would recommend doing this because you can damage the case. You won't see anything interesting there, everything is in the photo.

**Comment by ydzhus** [comment]
The new 2023 model has newer components. More reliable fastening of wires.
